ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/SelectionUtil.java

```diff
@@ -208,4 +216,99 @@ private static Stream<SelectExpression> resolveAllColumns(
             name
         )));
   }
+
+  private static Stream<SelectExpression> resolveStructAll(
+      final int idx,
+      final PlanNode parentNode,
+      final StructAll structAll,
+      final Optional<LogicalSchema> targetSchema
+  ) {
+    final LogicalSchema parentSchema = parentNode.getSchema();
+
+    // Resolve the base struct first. I don't know if this is necessary. Let's make it behave
+    // like a single column first.
+    final Stream<SelectExpression> selectExpression = resolveSingleColumn(
+        idx,
+        parentNode,
+        new SingleColumn(
+            structAll.getBaseStruct(),
+            Optional.of(ColumnName.of("UNUSED"))
+        ),
+        targetSchema
+    );
+
+    // The selectExpression produced above should return one single expression
+    final Expression expression = Streams.findLast(selectExpression).get().getExpression();
+
+    // Find all fields of the most nested struct referenced in the expression.
+    // i.e. `col0->l1->l2->*` should resolve all fields of `col0->l1->f2`
+    final SqlStruct mostNestedStructFields = resolveStructFields(parentSchema, expression);
+
+    // Now convert all fields to a SelectExpression
+    return mostNestedStructFields.fields().stream()
+        .map(f -> SelectExpression.of(
+            ColumnName.of(f.name()),
+            new DereferenceExpression(
+                Optional.empty(),
+                expression,
+                f.name().toUpperCase()
+            )));
+  }
+
+  private static SqlStruct resolveStructFields(
+      final LogicalSchema parentSchema,
+      final Expression expression
+  ) {
+    // Different expressions are accepted in this method.
+    //
+    // 1. DereferenceExpression. Contains struct columns with -> references. i.e. `col0->l1`
+    // 2. ColumnReferenceExp. Contains column references only. i.e. `col0`
+    //
+    // This resolver goes back to the column reference, so it has access to the column schema. It
+    // then goes forward each struct field, specified in the DereferenceExpression, to get
+    // access to the nested fields.
+
+    // This is the column part of the parent schema, which is used to access the column type
+    if (expression instanceof ColumnReferenceExp) {
```

Comment:
Ok, so are qualified and unqualified reference expressions handled ok in here?

Reply:
Yes, they're handled ok. Both expressions inherit from ColumnReferenceExp